NarrowTable.lua

```diff
@@ -0,0 +1,40 @@
+local NarrowTable, parent = torch.class('nn.NarrowTable', 'nn.Module')
+
+function NarrowTable:__init(offset, length)
+   parent.__init(self)
+   self.offset = offset
+   self.length = length or 1
+   if not offset then
+      error('nn.NarrowTable(offset, length)')
+   end
+   
+   self.output = {}
+   self.gradInput = {}
+end
+
+function NarrowTable:updateOutput(input)
+   for k,v in ipairs(self.output) do self.output[k] = v end
```

Comment:
Hmm yes this is indeed an error. should be = nil.

Nicholas Léonard
917-741-7570


On Wed, Jul 15, 2015 at 2:05 PM, Soumith Chintala <notifications@github.com>
wrote:

> In NarrowTable.lua
> <https://github.com/torch/nn/pull/317#discussion_r34709361>:
>
> > +local NarrowTable, parent = torch.class('nn.NarrowTable', 'nn.Module')
> > +
> > +function NarrowTable:__init(offset, length)
> > +   parent.__init(self)
> > +   self.offset = offset
> > +   self.length = length or 1
> > +   if not offset then
> > +      error('nn.NarrowTable(offset, length)')
> > +   end
> > +
> > +   self.output = {}
> > +   self.gradInput = {}
> > +end
> > +
> > +function NarrowTable:updateOutput(input)
> > +   for k,v in ipairs(self.output) do self.output[k] = v end
>
> hmm, I am very confused. You are just reassigning all the integer keys of
> self.output to their values. I dont see what has changed after this line,
> self.output is exactly the same.
>
> --
> Reply to this email directly or view it on GitHub
> <https://github.com/torch/nn/pull/317/files#r34709361>.
>